docs/include/attributes-ls.asciidoc

```diff
@@ -8,3 +8,275 @@ with a corresponding change to the VPR settings in
+
+:queuedrain: queue.drain
+:queuedrain-desc: When set to `true`, Logstash waits until the persistent queue is drained before shutting down. +
+The amount of time it takes to drain the queue depends on the number of events that have accumulated in the queue. +
+Tip: Avoid using this setting unless the queue, even when full, is relatively small and can be drained quickly. Default is `false`.
+
+:queuemaxevents: queue.max_events
+:queuemaxevents-desc: The maximum number of events not yet read by the pipeline worker (`queue.type: persisted`). The default is `0` (unlimited). +
+We use this setting for internal testing. +
+Users generally shouldn't change this value.
+
+:queuemaxbytes: queue.max_bytes
+:queuemaxbytes-desc: The total capacity of each queue in number of bytes (`queue.type: persisted`). Unless overridden in `pipelines.yml` or central management, each persistent queue will be sized at the value of `queue.max_bytes` specified in `logstash.yml`. +
+Be sure that your disk has sufficient capacity to handle the cumulative total of `queue.max_bytes` across all persistent queues. The total of `queue.max_bytes` for ALL queues should be lower than the capacity of your disk. +
+If both `queue.max_events` and `queue.max_bytes` are specified, Logstash uses whichever criteria is reached first. Default: 1024mb (1gb).
+
+:queuecheckpointacks: queue.checkpoint.acks 
+:queuecheckpointacks-desc: Sets the maximum number of acked events before forcing a checkpoint (`queue.type: persisted`). Default is 1024. Specify `queue.checkpoint.acks: 0` to set this value to unlimited.
+    
+:queuecheckpointwrites: queue.checkpoint.writes 
+:queuecheckpointwrites-desc: Sets the maximum number of written events before a forced checkpoint (`queue.type: persisted`). Default is 1024. Specify `queue.checkpoint.writes: 0` to set this value to unlimited. +
+To avoid losing data in the persistent queue, you can set `queue.checkpoint.writes: 1` to force a checkpoint after each event is written. +
+ Keep in mind that disk writes have a resource cost. Setting this value to 1 ensures maximum durability, but can severely impact performance. See <<durability-persistent-queues>> to better understand the trade-offs.
+
+:queuecheckpointretry: queue.checkpoint.retry
+:queuecheckpointretry-desc:	When enabled, Logstash will retry four times per attempted checkpoint write for any checkpoint writes that fail. Any subsequent errors are not retried. +
+This is a workaround for failed checkpoint writes that have been seen only on Windows platform, filesystems with non-standard behavior such as SANs and is not recommended except in those specific circumstances. (queue.type: persisted)+
+Default is `true`.
+
+:queuecheckpointinterval: queue.checkpoint.interval 
+:queuecheckpointinterval-desc: Sets the interval in milliseconds when a checkpoint is forced on the head page (`queue.type: persisted`). Default is 1000. Set to 0 to eliminate periodic checkpoints. 
+
+
+//DLQ settings
+
```

Comment:
Note to self @karenzone: `Add dead_letter_queue.retain.age` for both logstash.yml and pipelines.yml

Reply:
Included in #14384